nat_traversal/upnp.go

```diff
@@ -0,0 +1,54 @@
+package nat_traversal
+
+import (
+	"fmt"
+	"gitlab.com/NebulousLabs/go-upnp"
+)
+
+func GetExternalIp() (string, error) {
+	// connect to router
+	d, err := upnp.Discover()
+	if err != nil {
+		return "", fmt.Errorf("failed to connect to router: %v", err)
+	}
+
+	// discover external IP
+	ip, err := d.ExternalIP()
+	if err != nil {
+		return "", fmt.Errorf("failed to discover external IP: %v", err)
+	}
+
+	return ip, nil
+}
+
+func AcquirePort(port uint16) (uint16, error) {
+	// connect to router
+	d, err := upnp.Discover()
+	if err != nil {
+		return 0, fmt.Errorf("failed to connect to router: %v", err)
+	}
+
+	// forward a port
+	err = d.Forward(port, "upnp test")
```

Comment:
I'm not  really familiar with `go-upnp`'s API, though if the second param is a the service name then `upnp test` is not a good name..